pkg/arvo/sys/vane/gall.hoon

```diff
@@ -2,1457 +2,2882 @@
+  ==
+::
+::  +ducts: opaque input.
+::
+++  ducts
+  $:
+      :: bone sequence
+      ::
+      bone=@ud
+      :: by duct
+      ::
+      bone-map=(map duct bone)
+      :: by bone
+      ::
+      duct-map=(map bone duct)
+  ==
+::
+::  +misvale-data: subscribers with bad marks.
+::
+::    XX a hack, required to break a subscription loop
+::    which arises when an invalid mark crashes a diff.
+::    See usage in ap-misvale.
+::
+++  misvale-data  (set wire)
+::
+::  +agent: agent state.
+::
+++  agent
+  $:
+      :: bad reqs
+      ::
+      misvale=misvale-data
+      :: cache
+      ::
+      cache=worm
+      :: ap-find-arm cache
+      ::
+      arm-cache=(map [term path] (unit (pair @ud term)))
+      :: control duct
+      ::
+      control-duct=duct
+      :: unstopped
+      ::
+      live=?
+      :: privilege
+      ::
+      privilege=security-control
+      :: statistics
+      ::
+      stats=stats
+      :: subscribers
+      ::
+      subscribers=subscribers
+      :: running state
+      ::
+      running-state=vase
+      :: update control
+      ::
+      beak=beak
+      :: req'd translations
+      ::
+      marks=(map bone mark)
+      :: opaque ducts
+      ::
+      ducts=ducts
+  ==
+::
+:: +blocked: blocked tasks.
+::
+++  blocked  (qeu (trel duct privilege agent-action))
+::
+:: +stats: statistics.
+::
+++  stats
+  $:
+      :: change number
+      ::
+      change=@ud
+      :: entropy
+      ::
+      eny=@uvJ
+      :: time
+      ::
+      time=@da
+  ==
+--
+::
+:: (vane header)
+::
 .  ==
-=|  all/axle                                            ::  all vane state
-|=  $:  our=ship                                        ::  identity
-        now=@da                                         ::  urban time
-        eny=@uvJ                                        ::  entropy
-        ska=sley                                        ::  activate
-    ==                                                  ::  opaque core
+::
+:: (all vane state)
+::
+=|  =gall
+|=  $:
+        :: identity
+        ::
+        our=ship
+        :: urban time
+        ::
+        now=@da
+        :: entropy
+        ::
+        eny=@uvJ
+        :: activate
+        ::
+        ska=sley
+    ==
+::
+::  (opaque core)
+::
 ~%  %gall-top  ..is  ~
-|%  ::::::::::::::::::::::::::::::::::::::::::::::::::::::  state machine
-    ::::::::::::::::::::::::::::::::::::::::::::::::::::::
+::
+::  (state machine)
+::
+|%
+::
+::  +gall-payload:  gall payload.
+::
+++  gall-payload  +
+::
+::  +mo: Arvo-level move handling.
+::
+::  An outer core responsible for routing moves to and from Arvo; it calls
+::  an inner core, +ap, to route internal moves to and from agents.
+::
 ++  mo
   ~%  %gall-mo  +>  ~
-  =*  mas  mast.all
-  |_  $:  hen=duct
-          moz=(list move)
-      ==
-  ++  mo-abed                                           ::  initialize
+  ::
+  |_
+    $:
+      hen=duct
+      moves=(list move)
+    ==
+  ::
+  ++  mo-state  .
```

Comment:
We (or maybe just I? I think it's something of a convention) usually use 'state' to refer to a core's data, excluding the battery -- probably excluding the context too.  I got pretty confused when I saw +mo-boot emit a move but just produce 'state'.  I think 'state' usually has the connotation of concrete, clammable data, as distinct from cores, which include code.

This distinction is much sharper in Hoon than in Haskell because everything needs to hot-reload in Urbit and no one knows how to upgrade arbitrary closures, so we want the concrete data separate from the code when we reload.

This is the sort of naming convention we might want to document, although I don't think I knew it existed until just now.